internal/driver/driver.go

```diff
@@ -457,6 +459,12 @@ func (d *Driver) AddDevice(deviceName string, protocols map[string]models.Protoc
 	if err != nil {
 		return errors.NewCommonEdgeXWrapper(err)
 	}
+
+	device, err := d.sdkService.GetDeviceByName(deviceName)
+	if err != nil {
+		return errors.NewCommonEdgeXWrapper(err)
+	}
+	d.checkStatusOfDevice(device) // check the status of the newly added device
```

Comment:
Should the same be done for UpdateDevice() too?

Reply:
There may be a subset of cases of UpdateDevice where it makes sense to do this, but in general, this will cause an infinite loop as the checkStatus eventually will update the device metadata with status and last seen fields.